### DISCUSSION TITLE
Virtual bones is in wrong position for new poses after retargeting

### DISCUSSION THREAD

**Discussion by ameaninglessname** [Sixze/ALS-Refactored#197] [Q&A]
![image](https://user-images.githubusercontent.com/20263103/204816114-0dcf7b55-d86b-4a2b-8cd4-10367abdad0d.png)

This is a pose I retargeted from "BossyEnemy" pack on marketplace (original name was Boss_Attack_Swing_RM),
notice the virtual bone is de-sync with the relative foot bone.

However i think several ways could handle this:
1. delete all the virtual bones, re add them and restart the editor
2. write some functionality to refresh (re-calculate) the virtual bone information(did not look into it yet)
3. write a customized IK Rig for ALS skeleton for IK retargeter(retargeting process) to handle the virtual bone retargting / retargeted properly

Anyone got a better solution for this?

And do we need an IK retargeter for ALS?

**Comment by ve110cet** [comment]
I think that constraining (or reconstraining) the virtual bone might work? I think I had the same problem when I was working with Community.

This is what I did to fix it.  https://www.youtube.com/watch?v=8HsJBHDuCU8

**Reply by ameaninglessname** [reply]
Thanks for your suggestion, i see from the video that it need to modify and re-save all the animation, which is not good for me.

> delete all the virtual bones, re add them and restart the editor

If i do this👆, only the skeleton is modified, exist and newly re-targeted animation is leaving untouched.
